src/Backups/BackupIO_Disk.cpp

```diff
@@ -127,9 +130,9 @@ void BackupWriterDisk::copyFileNative(DiskPtr src_disk, const String & src_file_
     if (!src_disk)
         throw Exception(ErrorCodes::LOGICAL_ERROR, "Cannot natively copy data to disk without source disk");
 
-    if ((src_offset != 0) || (src_size != src_disk->getFileSize(src_file_name)))
+    if (has_throttling || (src_offset != 0) || (src_size != src_disk->getFileSize(src_file_name)))
     {
-        auto create_read_buffer = [src_disk, src_file_name] { return src_disk->readFile(src_file_name); };
+        auto create_read_buffer = [this, src_disk, src_file_name] { return src_disk->readFile(src_file_name, read_settings); };
```

Comment:
`read_settings` has two throttlers now (remote & local) - which one is going to be used here?

Reply:
It depends on the file that will be read, if remote file then `remote_throttler`, local - `local_throttler`

Reply:
And where is the code which checks that?